torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp

```diff
@@ -227,9 +263,44 @@ void insertDynamicShapesGuard(
     auto te_dyn_group = SubgraphUtils::createSingletonSubgraph(
         typecheck_node, prim::TensorExprDynamicGroup);
     SubgraphUtils::mergeNodeIntoSubgraph(versioning_if, te_dyn_group);
+    inlineFallbackGraphAndAddSRCopyOutOp(
+        SubgraphUtils::getSubgraph(te_dyn_group));
   }
 }
 
+// If there are tensors on the stack other than
+// the inputs, copies the inputs into the values
+// on the stack. Only used in SR.
+Operation StaticRuntimeCopyOuts(const Node* node) {
+  auto num_ten_inputs = node->inputs().size();
+  return [num_ten_inputs](Stack& stack) {
+    std::vector<IValue> inputs = pop(stack, num_ten_inputs);
+    // uncommon case - first run
+    if (stack.size() == 0) {
+      for (IValue elem : inputs) {
+        push(stack, std::move(elem));
+      }
+    } else {
+      at::ArrayRef<IValue> outputs = last(stack, num_ten_inputs);
```

Comment:
Why are we getting the last #inputs from the stack here? Shouldn't we getting the last #outputs instead?

Reply:
#inputs == #outputs. See code comment I added inline

Reply:
Thanks for the comments. This makes sense now :)